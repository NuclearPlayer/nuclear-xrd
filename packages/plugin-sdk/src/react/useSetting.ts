import { useEffect, useMemo, useRef, useState } from 'react';

import type { SettingsHost, SettingValue } from '../types/settings';

export const useSetting = <T extends SettingValue = SettingValue>(
  host: SettingsHost | undefined,
  id: string,
) => {
  const hostRef = useRef(host);
  hostRef.current = host;

  const [currentValue, setCurrentValue] = useState<T | undefined>(undefined);

  useEffect(() => {
    if (!hostRef.current) return;
    let unsubscribe: (() => void) | undefined;
    let isMounted = true;

    hostRef.current.get<T>(id).then((initialValue) => {
      if (!isMounted) return;
      setCurrentValue(initialValue);
      unsubscribe = hostRef.current?.subscribe<T>(id, (nextValue) =>
        setCurrentValue(nextValue),
      );
    });

    return () => {
      isMounted = false;
      if (unsubscribe) unsubscribe();
    };
  }, [id, hostRef]);

  const setValue = useMemo(
    () => (nextValue: T) => {
      if (!hostRef.current) return;
      void hostRef.current.set<T>(id, nextValue);
    },
    [id],
  );

  return [currentValue, setValue] as const;
};
