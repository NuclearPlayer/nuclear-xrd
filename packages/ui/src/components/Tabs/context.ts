import { createContext, useContext } from 'react';

type TabsContextValue = {
  ids: {
    groupId: string;
    listId: string;
    panelsId: string;
  };
  selectedIndex: number;
  vertical?: boolean;
  manual?: boolean;
  listClassName?: string;
  tabClassName?: string;
  panelsClassName?: string;
  panelClassName?: string;
};

export const TabsContext = createContext<TabsContextValue | null>(null);

export const useTabsContext = () => {
  const ctx = useContext(TabsContext);
  if (!ctx) {
    throw new Error('Tabs.* must be used within <Tabs.Root>');
  }
  return ctx;
};
