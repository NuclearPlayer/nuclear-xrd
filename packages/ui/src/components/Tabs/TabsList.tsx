import { TabList } from '@headlessui/react';
import { FC, PropsWithChildren } from 'react';

import { cn } from '../../utils';
import { useTabsContext } from './context';

type TabsListProps = PropsWithChildren<{
  className?: string;
}>;

export const TabsList: FC<TabsListProps> = ({ children, className }) => {
  const {
    ids: { listId },
    listClassName,
  } = useTabsContext();
  return (
    <TabList
      id={listId}
      className={cn('flex w-full items-center gap-2', listClassName, className)}
    >
      {children}
    </TabList>
  );
};
