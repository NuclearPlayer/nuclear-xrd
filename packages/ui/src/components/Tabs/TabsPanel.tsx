import { TabPanel } from '@headlessui/react';
import { FC, PropsWithChildren } from 'react';

import { cn } from '../../utils';
import { useTabsContext } from './context';

type TabsPanelProps = PropsWithChildren<{
  className?: string;
}>;

export const TabsPanel: FC<TabsPanelProps> = ({ children, className }) => {
  const { panelClassName } = useTabsContext();
  return (
    <TabPanel className={cn('outline-none', panelClassName, className)}>
      {children}
    </TabPanel>
  );
};
