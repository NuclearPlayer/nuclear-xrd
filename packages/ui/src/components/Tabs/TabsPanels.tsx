import { TabPanels } from '@headlessui/react';
import { FC, PropsWithChildren } from 'react';

import { cn } from '../../utils';
import { useTabsContext } from './context';

type TabsPanelsProps = PropsWithChildren<{
  className?: string;
}>;

export const TabsPanels: FC<TabsPanelsProps> = ({ children, className }) => {
  const {
    ids: { panelsId },
    panelsClassName,
  } = useTabsContext();
  return (
    <TabPanels
      id={panelsId}
      className={cn(
        'mt-2 flex flex-1 flex-col overflow-hidden',
        panelsClassName,
        className,
      )}
    >
      {children}
    </TabPanels>
  );
};
