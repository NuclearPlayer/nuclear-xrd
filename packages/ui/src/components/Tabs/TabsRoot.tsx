import { TabGroup } from '@headlessui/react';
import { FC, PropsWithChildren, useId } from 'react';

import { cn } from '../../utils';
import { TabsContext } from './context';

type TabsRootProps = PropsWithChildren<{
  defaultIndex?: number;
  selectedIndex?: number;
  onChange?: (index: number) => void;
  vertical?: boolean;
  manual?: boolean;
  className?: string;
  listClassName?: string;
  tabClassName?: string;
  panelsClassName?: string;
  panelClassName?: string;
}>;

export const TabsRoot: FC<TabsRootProps> = ({
  children,
  defaultIndex,
  selectedIndex,
  onChange,
  vertical,
  manual,
  className,
  listClassName,
  tabClassName,
  panelsClassName,
  panelClassName,
}) => {
  const reactId = useId();
  const groupId = `tabs-${reactId}`;
  const listId = `${groupId}-list`;
  const panelsId = `${groupId}-panels`;

  return (
    <TabGroup
      as="div"
      className={cn('flex w-full flex-col gap-2', className)}
      defaultIndex={selectedIndex === undefined ? defaultIndex : undefined}
      selectedIndex={selectedIndex}
      onChange={onChange}
      vertical={vertical}
      manual={manual}
    >
      {({ selectedIndex: idx }) => (
        <TabsContext.Provider
          value={{
            ids: { groupId, listId, panelsId },
            selectedIndex: idx,
            vertical,
            manual,
            listClassName,
            tabClassName,
            panelsClassName,
            panelClassName,
          }}
        >
          {children}
        </TabsContext.Provider>
      )}
    </TabGroup>
  );
};
