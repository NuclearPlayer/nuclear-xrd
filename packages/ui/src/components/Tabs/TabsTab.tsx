import { Tab } from '@headlessui/react';
import { FC, Fragment, PropsWithChildren } from 'react';

import { cn } from '../../utils';
import { Button } from '../Button';
import { useTabsContext } from './context';

type TabsTabProps = PropsWithChildren<{
  disabled?: boolean;
  className?: string;
}>;

export const TabsTab: FC<TabsTabProps> = ({
  children,
  disabled,
  className,
}) => {
  const { tabClassName } = useTabsContext();
  return (
    <Tab as={Fragment} disabled={disabled}>
      {({ selected }) => (
        <Button
          type="button"
          disabled={disabled}
          variant={selected ? 'default' : 'text'}
          className={cn(tabClassName, className)}
          data-selected={selected ? '' : undefined}
        >
          {children}
        </Button>
      )}
    </Tab>
  );
};
