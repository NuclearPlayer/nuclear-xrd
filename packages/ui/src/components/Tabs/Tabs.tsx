import { FC, ReactNode } from 'react';

import { TabsList } from './TabsList';
import { TabsPanel } from './TabsPanel';
import { TabsPanels } from './TabsPanels';
import { TabsRoot } from './TabsRoot';
import { TabsTab } from './TabsTab';

export type TabsItem = {
  id: string;
  label: ReactNode;
  content: ReactNode;
  disabled?: boolean;
};

export type TabsProps = {
  items?: TabsItem[];
  defaultIndex?: number;
  selectedIndex?: number;
  onChange?: (index: number) => void;
  vertical?: boolean;
  manual?: boolean;
  className?: string;
  listClassName?: string;
  tabClassName?: string;
  panelsClassName?: string;
  panelClassName?: string;
};

type TabsComponent = FC<TabsProps> & {
  Root: typeof TabsRoot;
  List: typeof TabsList;
  Tab: typeof TabsTab;
  Panels: typeof TabsPanels;
  Panel: typeof TabsPanel;
};

const TabsImpl: FC<TabsProps> = ({
  items,
  defaultIndex,
  selectedIndex,
  onChange,
  vertical,
  manual,
  className,
  listClassName,
  tabClassName,
  panelsClassName,
  panelClassName,
}) => {
  if (!items || items.length === 0) {
    // Composition mode only â€” consumer uses Tabs.Root + subcomponents
    return (
      <TabsRoot
        defaultIndex={defaultIndex}
        selectedIndex={selectedIndex}
        onChange={onChange}
        vertical={vertical}
        manual={manual}
        className={className}
        listClassName={listClassName}
        tabClassName={tabClassName}
        panelsClassName={panelsClassName}
        panelClassName={panelClassName}
      />
    );
  }

  return (
    <TabsRoot
      defaultIndex={defaultIndex}
      selectedIndex={selectedIndex}
      onChange={onChange}
      vertical={vertical}
      manual={manual}
      className={className}
      listClassName={listClassName}
      tabClassName={tabClassName}
      panelsClassName={panelsClassName}
      panelClassName={panelClassName}
    >
      <TabsList>
        {items.map((it) => (
          <TabsTab key={it.id} disabled={it.disabled}>
            {it.label}
          </TabsTab>
        ))}
      </TabsList>
      <TabsPanels>
        {items.map((it) => (
          <TabsPanel key={it.id}>{it.content}</TabsPanel>
        ))}
      </TabsPanels>
    </TabsRoot>
  );
};

export const Tabs = TabsImpl as TabsComponent;
Tabs.Root = TabsRoot;
Tabs.List = TabsList;
Tabs.Tab = TabsTab;
Tabs.Panels = TabsPanels;
Tabs.Panel = TabsPanel;
