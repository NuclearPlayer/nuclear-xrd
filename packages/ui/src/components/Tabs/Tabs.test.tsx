import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Box, Card, CardGrid, Loader } from '..';
import { Tabs } from './Tabs';

const ITEMS = [
  {
    id: 'artists',
    label: 'Artists',
    content: (
      <div className="text-foreground">
        <CardGrid>
          {[
            'Frank Zappa',
            'Ozzy Osbourne',
            'David Bowie',
            'Kurt Cobain',
            'John Lennon',
          ].map((item) => (
            <Card
              key={item}
              title={item}
              src={`https://picsum.photos/seed/${item}/300`}
            />
          ))}
        </CardGrid>
      </div>
    ),
  },
  {
    id: 'tracks',
    label: 'Tracks',
    content: (
      <div className="text-foreground">
        <Loader /> Still loading!
      </div>
    ),
  },
  {
    id: 'about',
    label: 'About',
    content: (
      <div className="text-foreground">
        <Box>Section in a box.</Box>
      </div>
    ),
  },
];

describe('Tabs', () => {
  it('(Snapshot) renders correctly', () => {
    const { asFragment } = render(<Tabs items={ITEMS} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('(Snapshot) switches tabs', async () => {
    const { asFragment } = render(<Tabs items={ITEMS} />);
    await userEvent.click(await screen.findByText('About'));
    expect(asFragment()).toMatchSnapshot();
  });
});
