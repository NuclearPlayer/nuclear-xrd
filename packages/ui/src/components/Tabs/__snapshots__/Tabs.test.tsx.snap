// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabs > (Snapshot) renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex max-h-full w-full flex-col gap-2"
  >
    <div
      aria-orientation="horizontal"
      class="flex w-full items-center gap-2"
      id="tabs-:r0:-list"
      role="tablist"
    >
      <button
        aria-controls="headlessui-tabs-panel-:r7:"
        aria-selected="true"
        class="inline-flex cursor-pointer items-center rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-primary border-border shadow-shadow hover:translate-x-shadow-x hover:translate-y-shadow-y border-2 hover:shadow-none h-10 px-4 py-2"
        data-headlessui-state=""
        data-selected=""
        id="headlessui-tabs-tab-:r1:"
        role="tab"
        tabindex="0"
        type="button"
      >
        Artists
      </button>
      <button
        aria-controls="headlessui-tabs-panel-:r9:"
        aria-selected="false"
        class="inline-flex cursor-pointer items-center rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-transparent h-10 px-4 py-2"
        data-headlessui-state=""
        id="headlessui-tabs-tab-:r3:"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tracks
      </button>
      <button
        aria-controls="headlessui-tabs-panel-:rb:"
        aria-selected="false"
        class="inline-flex cursor-pointer items-center rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-transparent h-10 px-4 py-2"
        data-headlessui-state=""
        id="headlessui-tabs-tab-:r5:"
        role="tab"
        tabindex="-1"
        type="button"
      >
        About
      </button>
    </div>
    <div
      class="mt-2 flex flex-1 flex-col overflow-hidden"
      id="tabs-:r0:-panels"
    >
      <div
        aria-labelledby="headlessui-tabs-tab-:r1:"
        class="relative flex-1 overflow-hidden outline-none"
        data-headlessui-state="selected"
        data-selected=""
        id="headlessui-tabs-panel-:r7:"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="relative h-full w-full"
        >
          <div
            class="scrollbar-hide flex h-full w-full flex-col overflow-auto pr-6 pb-6"
            style="scrollbar-width: none;"
          >
            <div
              class="text-foreground"
            >
              <div
                class="grid grid-cols-[repeat(auto-fit,minmax(10rem,1fr))] gap-4"
                role="grid"
              >
                <button
                  class="cursor-pointer rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-primary border-border shadow-shadow hover:translate-x-shadow-x hover:translate-y-shadow-y border-2 hover:shadow-none h-auto flex w-42 flex-col items-stretch gap-2 p-2 text-left"
                  data-headlessui-state=""
                  data-testid="card"
                  type="submit"
                >
                  <div
                    class="border-border text-foreground flex h-full rounded border-2 bg-primary shadow-none relative aspect-square w-full items-center justify-center overflow-hidden p-0"
                  >
                    <div
                      class="overflow-hidden absolute inset-0"
                    >
                      <img
                        alt="Frank Zappa"
                        class="h-full w-full object-cover"
                        decoding="async"
                        loading="lazy"
                        src="https://picsum.photos/seed/Frank Zappa/300"
                      />
                      <div
                        class="pointer-events-none absolute inset-0 flex items-center justify-center"
                      >
                        <svg
                          aria-hidden="true"
                          class="lucide lucide-cassette-tape animate-pulse opacity-70"
                          fill="none"
                          height="96"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="0.5"
                          viewBox="0 0 24 24"
                          width="96"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            height="16"
                            rx="2"
                            width="20"
                            x="2"
                            y="4"
                          />
                          <circle
                            cx="8"
                            cy="10"
                            r="2"
                          />
                          <path
                            d="M8 12h8"
                          />
                          <circle
                            cx="16"
                            cy="10"
                            r="2"
                          />
                          <path
                            d="m6 20 .7-2.9A1.4 1.4 0 0 1 8.1 16h7.8a1.4 1.4 0 0 1 1.4 1l.7 3"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    class="min-w-0"
                  >
                    <div
                      class="text-foreground truncate text-sm font-bold"
                    >
                      Frank Zappa
                    </div>
                  </div>
                </button>
                <button
                  class="cursor-pointer rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-primary border-border shadow-shadow hover:translate-x-shadow-x hover:translate-y-shadow-y border-2 hover:shadow-none h-auto flex w-42 flex-col items-stretch gap-2 p-2 text-left"
                  data-headlessui-state=""
                  data-testid="card"
                  type="submit"
                >
                  <div
                    class="border-border text-foreground flex h-full rounded border-2 bg-primary shadow-none relative aspect-square w-full items-center justify-center overflow-hidden p-0"
                  >
                    <div
                      class="overflow-hidden absolute inset-0"
                    >
                      <img
                        alt="Ozzy Osbourne"
                        class="h-full w-full object-cover"
                        decoding="async"
                        loading="lazy"
                        src="https://picsum.photos/seed/Ozzy Osbourne/300"
                      />
                      <div
                        class="pointer-events-none absolute inset-0 flex items-center justify-center"
                      >
                        <svg
                          aria-hidden="true"
                          class="lucide lucide-cassette-tape animate-pulse opacity-70"
                          fill="none"
                          height="96"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="0.5"
                          viewBox="0 0 24 24"
                          width="96"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            height="16"
                            rx="2"
                            width="20"
                            x="2"
                            y="4"
                          />
                          <circle
                            cx="8"
                            cy="10"
                            r="2"
                          />
                          <path
                            d="M8 12h8"
                          />
                          <circle
                            cx="16"
                            cy="10"
                            r="2"
                          />
                          <path
                            d="m6 20 .7-2.9A1.4 1.4 0 0 1 8.1 16h7.8a1.4 1.4 0 0 1 1.4 1l.7 3"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    class="min-w-0"
                  >
                    <div
                      class="text-foreground truncate text-sm font-bold"
                    >
                      Ozzy Osbourne
                    </div>
                  </div>
                </button>
                <button
                  class="cursor-pointer rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-primary border-border shadow-shadow hover:translate-x-shadow-x hover:translate-y-shadow-y border-2 hover:shadow-none h-auto flex w-42 flex-col items-stretch gap-2 p-2 text-left"
                  data-headlessui-state=""
                  data-testid="card"
                  type="submit"
                >
                  <div
                    class="border-border text-foreground flex h-full rounded border-2 bg-primary shadow-none relative aspect-square w-full items-center justify-center overflow-hidden p-0"
                  >
                    <div
                      class="overflow-hidden absolute inset-0"
                    >
                      <img
                        alt="David Bowie"
                        class="h-full w-full object-cover"
                        decoding="async"
                        loading="lazy"
                        src="https://picsum.photos/seed/David Bowie/300"
                      />
                      <div
                        class="pointer-events-none absolute inset-0 flex items-center justify-center"
                      >
                        <svg
                          aria-hidden="true"
                          class="lucide lucide-cassette-tape animate-pulse opacity-70"
                          fill="none"
                          height="96"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="0.5"
                          viewBox="0 0 24 24"
                          width="96"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            height="16"
                            rx="2"
                            width="20"
                            x="2"
                            y="4"
                          />
                          <circle
                            cx="8"
                            cy="10"
                            r="2"
                          />
                          <path
                            d="M8 12h8"
                          />
                          <circle
                            cx="16"
                            cy="10"
                            r="2"
                          />
                          <path
                            d="m6 20 .7-2.9A1.4 1.4 0 0 1 8.1 16h7.8a1.4 1.4 0 0 1 1.4 1l.7 3"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    class="min-w-0"
                  >
                    <div
                      class="text-foreground truncate text-sm font-bold"
                    >
                      David Bowie
                    </div>
                  </div>
                </button>
                <button
                  class="cursor-pointer rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-primary border-border shadow-shadow hover:translate-x-shadow-x hover:translate-y-shadow-y border-2 hover:shadow-none h-auto flex w-42 flex-col items-stretch gap-2 p-2 text-left"
                  data-headlessui-state=""
                  data-testid="card"
                  type="submit"
                >
                  <div
                    class="border-border text-foreground flex h-full rounded border-2 bg-primary shadow-none relative aspect-square w-full items-center justify-center overflow-hidden p-0"
                  >
                    <div
                      class="overflow-hidden absolute inset-0"
                    >
                      <img
                        alt="Kurt Cobain"
                        class="h-full w-full object-cover"
                        decoding="async"
                        loading="lazy"
                        src="https://picsum.photos/seed/Kurt Cobain/300"
                      />
                      <div
                        class="pointer-events-none absolute inset-0 flex items-center justify-center"
                      >
                        <svg
                          aria-hidden="true"
                          class="lucide lucide-cassette-tape animate-pulse opacity-70"
                          fill="none"
                          height="96"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="0.5"
                          viewBox="0 0 24 24"
                          width="96"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            height="16"
                            rx="2"
                            width="20"
                            x="2"
                            y="4"
                          />
                          <circle
                            cx="8"
                            cy="10"
                            r="2"
                          />
                          <path
                            d="M8 12h8"
                          />
                          <circle
                            cx="16"
                            cy="10"
                            r="2"
                          />
                          <path
                            d="m6 20 .7-2.9A1.4 1.4 0 0 1 8.1 16h7.8a1.4 1.4 0 0 1 1.4 1l.7 3"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    class="min-w-0"
                  >
                    <div
                      class="text-foreground truncate text-sm font-bold"
                    >
                      Kurt Cobain
                    </div>
                  </div>
                </button>
                <button
                  class="cursor-pointer rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-primary border-border shadow-shadow hover:translate-x-shadow-x hover:translate-y-shadow-y border-2 hover:shadow-none h-auto flex w-42 flex-col items-stretch gap-2 p-2 text-left"
                  data-headlessui-state=""
                  data-testid="card"
                  type="submit"
                >
                  <div
                    class="border-border text-foreground flex h-full rounded border-2 bg-primary shadow-none relative aspect-square w-full items-center justify-center overflow-hidden p-0"
                  >
                    <div
                      class="overflow-hidden absolute inset-0"
                    >
                      <img
                        alt="John Lennon"
                        class="h-full w-full object-cover"
                        decoding="async"
                        loading="lazy"
                        src="https://picsum.photos/seed/John Lennon/300"
                      />
                      <div
                        class="pointer-events-none absolute inset-0 flex items-center justify-center"
                      >
                        <svg
                          aria-hidden="true"
                          class="lucide lucide-cassette-tape animate-pulse opacity-70"
                          fill="none"
                          height="96"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="0.5"
                          viewBox="0 0 24 24"
                          width="96"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            height="16"
                            rx="2"
                            width="20"
                            x="2"
                            y="4"
                          />
                          <circle
                            cx="8"
                            cy="10"
                            r="2"
                          />
                          <path
                            d="M8 12h8"
                          />
                          <circle
                            cx="16"
                            cy="10"
                            r="2"
                          />
                          <path
                            d="m6 20 .7-2.9A1.4 1.4 0 0 1 8.1 16h7.8a1.4 1.4 0 0 1 1.4 1l.7 3"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div
                    class="min-w-0"
                  >
                    <div
                      class="text-foreground truncate text-sm font-bold"
                    >
                      John Lennon
                    </div>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        aria-hidden="true"
        aria-labelledby="headlessui-tabs-tab-:r3:"
        id="headlessui-tabs-panel-:r9:"
        role="tabpanel"
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
        tabindex="-1"
      />
      <span
        aria-hidden="true"
        aria-labelledby="headlessui-tabs-tab-:r5:"
        id="headlessui-tabs-panel-:rb:"
        role="tabpanel"
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
        tabindex="-1"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tabs > (Snapshot) renders in composition mode 1`] = `
<DocumentFragment>
  <div
    class="flex max-h-full w-full flex-col gap-2"
  >
    <div
      aria-orientation="horizontal"
      class="flex w-full items-center gap-2"
      id="tabs-:rq:-list"
      role="tablist"
    >
      <button
        aria-controls="headlessui-tabs-panel-:rv:"
        aria-selected="true"
        class="inline-flex cursor-pointer items-center rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-primary border-border shadow-shadow hover:translate-x-shadow-x hover:translate-y-shadow-y border-2 hover:shadow-none h-10 px-4 py-2"
        data-headlessui-state=""
        data-selected=""
        id="headlessui-tabs-tab-:rr:"
        role="tab"
        tabindex="0"
        type="button"
      >
        First
      </button>
      <button
        aria-controls="headlessui-tabs-panel-:r11:"
        aria-selected="false"
        class="inline-flex cursor-pointer items-center rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-transparent h-10 px-4 py-2"
        data-headlessui-state=""
        id="headlessui-tabs-tab-:rt:"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Second
      </button>
    </div>
    <div
      class="mt-2 flex flex-1 flex-col overflow-hidden"
      id="tabs-:rq:-panels"
    >
      <div
        aria-labelledby="headlessui-tabs-tab-:rr:"
        class="relative flex-1 overflow-hidden outline-none"
        data-headlessui-state="selected"
        data-selected=""
        id="headlessui-tabs-panel-:rv:"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="relative h-full w-full"
        >
          <div
            class="scrollbar-hide flex h-full w-full flex-col overflow-auto pr-6 pb-6"
            style="scrollbar-width: none;"
          >
            <div>
              First panel content
            </div>
          </div>
        </div>
      </div>
      <span
        aria-hidden="true"
        aria-labelledby="headlessui-tabs-tab-:rt:"
        id="headlessui-tabs-panel-:r11:"
        role="tabpanel"
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
        tabindex="-1"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tabs > (Snapshot) switches tabs 1`] = `
<DocumentFragment>
  <div
    class="flex max-h-full w-full flex-col gap-2"
  >
    <div
      aria-orientation="horizontal"
      class="flex w-full items-center gap-2"
      id="tabs-:rd:-list"
      role="tablist"
    >
      <button
        aria-controls="headlessui-tabs-panel-:rk:"
        aria-selected="false"
        class="inline-flex cursor-pointer items-center rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-transparent h-10 px-4 py-2"
        data-headlessui-state=""
        id="headlessui-tabs-tab-:re:"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Artists
      </button>
      <button
        aria-controls="headlessui-tabs-panel-:rm:"
        aria-selected="false"
        class="inline-flex cursor-pointer items-center rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-transparent h-10 px-4 py-2"
        data-headlessui-state=""
        id="headlessui-tabs-tab-:rg:"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tracks
      </button>
      <button
        aria-controls="headlessui-tabs-panel-:ro:"
        aria-selected="true"
        class="inline-flex cursor-pointer items-center rounded transition-all disabled:cursor-not-allowed disabled:opacity-50 text-foreground bg-primary border-border shadow-shadow hover:translate-x-shadow-x hover:translate-y-shadow-y border-2 hover:shadow-none h-10 px-4 py-2"
        data-headlessui-state="hover"
        data-hover=""
        data-selected=""
        id="headlessui-tabs-tab-:ri:"
        role="tab"
        tabindex="0"
        type="button"
      >
        About
      </button>
    </div>
    <div
      class="mt-2 flex flex-1 flex-col overflow-hidden"
      id="tabs-:rd:-panels"
    >
      <span
        aria-hidden="true"
        aria-labelledby="headlessui-tabs-tab-:re:"
        id="headlessui-tabs-panel-:rk:"
        role="tabpanel"
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
        tabindex="-1"
      />
      <span
        aria-hidden="true"
        aria-labelledby="headlessui-tabs-tab-:rg:"
        id="headlessui-tabs-panel-:rm:"
        role="tabpanel"
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
        tabindex="-1"
      />
      <div
        aria-labelledby="headlessui-tabs-tab-:ri:"
        class="relative flex-1 overflow-hidden outline-none"
        data-headlessui-state="selected"
        data-selected=""
        id="headlessui-tabs-panel-:ro:"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="relative h-full w-full"
        >
          <div
            class="scrollbar-hide flex h-full w-full flex-col overflow-auto pr-6 pb-6"
            style="scrollbar-width: none;"
          >
            <div
              class="text-foreground"
            >
              <div
                class="border-border text-foreground flex h-full w-full rounded border-2 p-4 bg-primary shadow-shadow"
              >
                Section in a box.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
